lib/OAuth2.php

```diff
@@ -405,6 +413,20 @@ public function __construct(IOAuth2Storage $storage, $config = array())
         }
     }
 
+    /**
+     * Sets a TokenGenerator to override default implementation.
+     *
+     * @param TokenGeneratorInterface $tokenGenerator
+     *
+     * @return OAuth2 The application (for chained calls of this method)
+     */
+    public function setTokenGenerator(TokenGeneratorInterface $tokenGenerator)
+    {
+        $this->tokenGenerator = $tokenGenerator;
```

Comment:
The DI container should inject the specific generator so the value is set.
If no specific generator is available, then the default generator should be used.